components/places/tests/fennec_bookmarks.rs

```diff
@@ -466,3 +459,139 @@ fn test_empty_db() -> Result<()> {
+    //         remote_node: Node(Tree { entry_index_by_guid: {Guid("menu________"): 2, Guid("toolbar_____"): 4, Guid("bookmarkAAAA"): 1, Guid("unfiled_____"): 5, Guid("root________"): 0, Guid("mobile______"): 3}, entries: [TreeEntry { item: Item { guid: Guid("root________"), kind: Folder, age: 1579567545000, needs_merge: false, validity: Valid }, content: None, divergence: Consistent, parent_index: None, child_indices: [2, 4, 5, 3] }, TreeEntry { item: Item { guid: Guid("bookmarkAAAA"), kind: Bookmark, age: 10000, needs_merge: true, validity: Valid }, content: Some(Bookmark { title: "A", url_href: "http://example.com/a/%s" }), divergence: Consistent, parent_index: Some(5), child_indices: [] }, TreeEntry { item: Item { guid: Guid("menu________"), kind: Folder, age: 1579567545000, needs_merge: false, validity: Valid }, content: None, divergence: Consistent, parent_index: Some(0), child_indices: [] }, TreeEntry { item: Item { guid: Guid("mobile______"), kind: Folder, age: 1579567545000, needs_merge: false, validity: Valid }, content: None, divergence: Consistent, parent_index: Some(0), child_indices: [] }, TreeEntry { item: Item { guid: Guid("toolbar_____"), kind: Folder, age: 1579567545000, needs_merge: false, validity: Valid }, content: None, divergence: Consistent, parent_index: Some(0), child_indices: [] }, TreeEntry { item: Item { guid: Guid("unfiled_____"), kind: Folder, age: 10000, needs_merge: true, validity: Valid }, content: Some(Folder { title: "unfiled" }), divergence: Consistent, parent_index: Some(0), child_indices: [1] }], deleted_guids: {}, problems: Problems({}) }, TreeEntry { item: Item { guid: Guid("unfiled_____"), kind: Folder, age: 10000, needs_merge: true, validity: Valid }, content: Some(Folder { title: "unfiled" }), divergence: Consistent, parent_index: Some(0), child_indices: [1] }) }
+
+    // XXX - "bookmarkAAAA" is in outgoing when local is newer - *with* keyword
+    // with *without* tag - not clear why that difference.
```

Comment:
Oh hey, we talked about this too! 😁 Summarizing for posterity, it's because of how we store keywords. Since they're always read out of `moz_bookmarks_synced` (unlike tags, we don't have a local table for keywords, or APIs for changing them), we'll always use the remote keyword, regardless of which side is newer. We accidentally did the right thing! (For the future, #2501 shouldn't change this, because we'll store imported keywords in the local table—so, even if the imported bookmarks are newer, we'll still preserve the keyword).